### PR TITLE
Mirror Mode

### PR DESCRIPTION
**Pull request by garrettjoecox**
Putting this up to get some playtesting, probably plenty of things broken, will add them below as they are found:

## Todo:
- [x] Stars moving the wrong way
- [x] Minimap
- [x] Kaleido map

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2008942210.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2008948402.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2008969128.zip)

### CODE REVIEW

**Comment by sitton76** [comment]
Seems the stars in the sky move with the players camera when enabled.

https://github.com/HarbourMasters/2ship2harkinian/assets/58642183/5e6414bb-946f-4099-aa81-63102bdc0495

**Comment by SuperBro64** [comment]
Besides flipping the mini-maps, something that probably should also be added to the to-do list is flipping the Map Subscreen overworld map image, since SoH's Mirror Mode does that, too.

**Comment by garrettjoecox** [comment]
Rebased & fixed the stars. I don't have much desire to port the minimap & kaleido map flipping. This should be good to go unless those are blockers

**Review by Archez — APPROVED** [review]
Looks good to me! Considering that mini-maps aren't even extracted properly, I'm fine with this merging as is.

I'll be tackling HD minimap support later so I can add mirror mode to my list.

**Comment by balloondude2** [comment]
It seems the issues related to starting minigames mentioned in the Discord thread are still an issue. The user called out the clocktown archery, swordsman school, and milk bar song. When these start, Link tries to go to a specific location, but is unable to get there in mirror mode softlocking the game. Turning off the option will allow Link to get to the starting location.

**Comment by garrettjoecox** [comment]
> It seems the issues related to starting minigames mentioned in the Discord thread are still an issue.

Oh I didn't even see those in the thread, thanks for the heads up!

**Comment by Archez** [comment]
Looks like func `func_800B6F20`. Kafei also uses it, so probably should check if that also needs player vs kafei logic.

**Comment by garrettjoecox** [comment]
> Looks like func `func_800B6F20`. Kafei also uses it, so probably should check if that also needs player vs kafei logic.

Nice, adding the inversion there allows me to undo the `isRealPlayer` I had to add to `Lib_GetControlStickData` for kafei. 

Tested all of those scenarios and everything appears to be in order.

**Comment by Archez** [comment]
> Nice, adding the inversion there allows me to undo the `isRealPlayer` I had to add to `Lib_GetControlStickData` for kafei

🔥

File: mm/src/code/z_actor.c [hunk 1]
```diff
@@ -1402,6 +1402,10 @@ void func_800B6F20(PlayState* play, Input* input, f32 magnitude, s16 baseYaw) {
     s16 relativeYaw = baseYaw - Camera_GetInputDirYaw(GET_ACTIVE_CAM(play));
 
     input->cur.stick_x = -Math_SinS(relativeYaw) * magnitude;
+    // 2S2H [Enhancement] Allow inverting the X axis with GI, primarily for mirror mode,
```
Comment (Archez) [inline mm/src/code/z_actor.c]:
```suggestion
    // 2S2H [Enhancement] Allow inverting the X axis with GI, primarily for mirror mode,
```

**Comment by Archez** [comment]
Updated to include minimaps and pause maps, as well as a few more missing things like:
* Climbing X movement
* Telescope X movement
* Debug camera DPad left/right roll
* Debug noclip left/right movement
* Actor NameTags
* The ocarina song staff wall in Termina Field

**Pull request merged**